### DISCUSSION TITLE
DOMException: BodyStreamBuffer was aborted at m.destroy

### DISCUSSION THREAD

**Discussion by gregorio-disco** [katspaugh/wavesurfer.js#3739] [Q&A] [answered]
Hey! 👋 I'm currently working on an audio tool that can navigate between multiple audios if the navigation process is done quickly, I get an unhandled error by the library.

## Bug description
The library throws an unhandled exception if the `wavesurfer` instance is destroyed **(with .destroy() method)** while still loading or processing an audio.

## Environment
 - Browser: Chrome, Edge, Firefox

## Minimal code snippet
```
// somewhere.jsx
...
      if (player.current) {
        player.current.destroy();
      }

      player.current = WaveSurfer.create({
        container: histogramRef.current,
        plugins: [Zoom],
        ...options,
      });
 ...
```

## Expected result
A handled exception

## Obtained result
An unhandled exception

## Screenshots
![image-20240516-143814](https://github.com/katspaugh/wavesurfer.js/assets/152921787/44e22507-a297-43d0-b13f-2f9f721d8815)

**Comment by katspaugh** [comment, marked as answer]
You should catch the promise from the `load` method. E.g.

```js
// somewhere.jsx
...
      if (player.current) {
        player.current.destroy();
      }

      player.current = WaveSurfer.create({
        container: histogramRef.current,
        plugins: [Zoom],
        ...options, // don't pass a URL here
      });

      player.current.load(url).catch(() => {
        // handle errors here
      })
 ...
```

Also consider using https://github.com/katspaugh/wavesurfer-react
